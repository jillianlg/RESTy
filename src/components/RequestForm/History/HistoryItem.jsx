import React from 'react';
import PropTypes from 'prop-types';
const HistoryItem = ({ id, method, url }) => {
  return (
    <ul key={id}>
      <p key={id}>{method}</p>
      <span key={id}>{url}</span>
    </ul>
  );
};

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default HistoryItem;

