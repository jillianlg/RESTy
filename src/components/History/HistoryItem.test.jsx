import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  afterEach(() => cleanup());
  it('renders a history item', () => {
    const { asFragment } = render(
      <HistoryItem
        id="1"
        method="GET"
        url="wwww.url.com"
        onChange={jest.fn()}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
