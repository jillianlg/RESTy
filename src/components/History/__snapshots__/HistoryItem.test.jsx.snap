// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders a history item 1`] = `
<DocumentFragment>
  <ul
    id="1"
  >
    <p
      id="1"
    >
      GET
    </p>
    <span
      id="1"
    >
      wwww.url.com
    </span>
  </ul>
</DocumentFragment>
`;
